pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -182,6 +182,14 @@ public void init(SegmentGeneratorConfig segmentCreationSpec, SegmentIndexCreatio
           LOGGER.info("Creating dictionary index in column {}.{} even when it is disabled in config",
               segmentCreationSpec.getTableName(), columnName);
         }
+
+        // override dictionary type if configured to do so
+        if (_config.isOptimizeDictionaryType()) {
+          LOGGER.info("Overriding dictionary type for column: {} using var-length dictionary: {}", columnName,
+              columnIndexCreationInfo.isUseVarLengthDictionary());
+          dictConfig = new DictionaryIndexConfig(dictConfig, columnIndexCreationInfo.isUseVarLengthDictionary());
```

Comment:
what is the default dictionary type? shall we name it `OptimizeWithVarLengthDictionaryType`?

Reply:
It is whichever is specified in the table config, the addition option will ignore the static config and use column index creation info - it does not necessary go var -> fixed or fixed -> var, so I think naming should stay generic